website/content/docs/enterprise/ent-to-ce-downgrades.mdx

```diff
@@ -0,0 +1,112 @@
+```
+
+You can inspect the snapshot to ensure if was successful with:
+
+```
+consul snapshot inspect backup.snap
+```
+
+Example output:
+
+```
+ID           2-1182-1542056499724
+Size         4115
+Index        1182
+Term         2
+Version      1
+```
+
+This will ensure you have a safe fallback option in case something goes wrong. Store
+this snapshot somewhere safe. More documentation on snapshot usage is available here:
+
+- [consul.io/commands/snapshot](/consul/commands/snapshot)
+- [Backup Consul Data and State tutorial](/consul/tutorials/production-deploy/backup-and-restore)
+
+**2.** Temporarily modify your Consul configuration so that its [log_level](/consul/docs/agent/config/cli-flags#_log_level)
+is set to `debug`. After doing this, issue the following command on your servers to
+reload the configuration:
+
+```
+consul reload
+```
+
+This change will give you more information to work with in the event something goes wrong.
+
+
+**3.** Remove references to tenanted services from service-resolver, service-splitter and service-router config entries.
+
+
+## Perform the Downgrade
+
+1. Restart or redeploy all clients with a CE version of the binary
+
+2. Update server binaries to use the CE version
+
+3. Restart or redeploy the servers one at a time with an env var named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` set to `true` on the servers.
+
+3. Validate that all servers are on CE and healthy using `consul members` command
+
+## What happens during the downgrade
+
+During the downgrade as the raft replication logs start flowing into the ce server we handle them in one of the following ways:
+
+- Drop the request
+
+   For example :
+   - Register requests - register requests having non default namespace, having services or health checks in non default namespace or partition are dropped.
+
+   - Peering Write Requests - peering write requests are dropped if the local partiton connecting to the peer is non default. 
+
+- Filter out tenanted data from the request
+
+  For example :
+  - Intention Sources - Intention sources targeting non default namespaces or partitions are filtered out of the config entry.
+  - Exported Services - Exports of services within non default namespaces or partitions are filtered out of the config entry.
+
+- Panic and stop the downgrade
+
+  Config Entries regarding traffic routing cannot be safely filtered because Consul cannot know whether the resulting filtered config entries send traffic towards services which can handle that traffic. In these situations, you are required to first remove references to services within non default namespaces or partitions from within those config entries. When encountering these config entries Consul CE in the downgrade mode will panic in order to not cause harm to existing service mesh routing.
+  
+  For example :
+  - Service Splitter,Service Resolver,Service Router Config Entry requests - if these config entries have references to tenanted services it will cause a panic.
```

Comment:
```suggestion
## Downgrade process details

During the downgrade process, the Consul CE server handles the raft replication logs in one of the following ways:

- Drops the request. 
- Filters out data from requests sent from non-default namespaces and partitions
- Panics and stops the downgrade 

The server may drop the following types of requests: 

- Registration requests in non-default namespace and services or health checks in non-default namespaces or partitions.

- Write requests to peered clusters if the local partiton connecting to the peer is non-default. 

The server may filter out data from requests to non-default namespaces and partitions in the following cases:

- Intention sources that target non-default namespaces or partitions are filtered out of the configuration entry.
- Exports of services within non-default namespaces or partitions are filtered out of the configuration entry.

The server may panic and stop the downgrade when configuration entries that route traffic cannot be safely filtered. This is because Consul cannot know whether the resulting filtered configuration entries send traffic to services that can handle the traffic. Consul CE panics in order to prevent harm to existing service mesh routes.

In these situations, you must first remove references to services within non-default namespaces or partitions from those configuration entries. 
  
The server may panic in the following cases:

- Service splitter, service resolver, and service router configuration entry requests that have references to services located in non-default namespaces and partitions cause the server to panic.
```
Cleaned up some grammar and style guide inconsistencies. 
I think this section should appear immediately after the Introduction section. It is background information to help readers understand if they want to perform the procedure described in the topic. 
The term "tenanted" is sort of difficult to understand right away, so I changed it to "non-default namespaces and partitions". Is that what we mean? It's a little longer, but does not require as much interpretation.